mlflow/langchain/utils/chat.py

```diff
@@ -266,33 +287,35 @@ def transform_request_json_for_chat_if_necessary(request_json, lc_model):
             2. A boolean indicating whether or not the request was transformed from the OpenAI
             chat format.
     """
-    if not should_transform_requst_json_for_chat(lc_model):
-        return request_json, False
 
-    def json_dict_might_be_chat_request(json_message: dict):
+    def json_dict_might_be_chat_request(json_message):
         return (
             isinstance(json_message, dict)
             and "messages" in json_message
             and
             # Additional keys can't be specified when calling LangChain invoke() / batch()
             # with chat messages
             len(json_message) == 1
+            # messages field should be a list
+            and isinstance(json_message["messages"], list)
         )
 
-    if isinstance(request_json, dict) and json_dict_might_be_chat_request(request_json):
-        try:
-            return _convert_chat_request_or_throw(request_json), True
-        except pydantic.ValidationError:
-            return request_json, False
-    elif isinstance(request_json, list) and all(
-        json_dict_might_be_chat_request(json) for json in request_json
-    ):
+    def is_list_of_chat_messages(json_message: list[dict]):
+        return isinstance(json_message, list) and all(
+            json_dict_might_be_chat_request(message) for message in json_message
+        )
+
+    should_convert = MLFLOW_CONVERT_MESSAGES_DICT_TO_LIST_OF_BASEMESSAGES_FOR_LANGCHAIN.get()
+    if should_convert is None:
+        should_convert = should_transform_requst_json_for_chat(lc_model) and (
+            json_dict_might_be_chat_request(request_json) or is_list_of_chat_messages(request_json)
+        )
+
+    if should_convert:
         try:
-            return (
-                [_convert_chat_request_or_throw(json_dict) for json_dict in request_json],
-                True,
-            )
-        except pydantic.ValidationError:
+            return convert_chat_request(request_json), True
+        # we should catch all exceptions here including pydantic validation errors
+        except Exception:
```

Comment:
why catch any exceptions?

Reply:
Because I want to replace _convert_chat_request_or_throw to use langchain's convert_to_messages method (which could throw other errors) and finally drop _ChatRequest in this file (the current code means we need to maintain _ChatRequest)
But maybe I should drop this change given the risk of introducing regression because of this change